File: app/src/main/java/mx/yellowme/app/MainActivity.java
package mx.yellowme.app;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import mx.yellowme.app.simple.UmbrellaActivity;
import mx.yellowme.app.simple.UmbrellaFragment;

public class MainActivity extends AppCompatActivity {

    private Button example1;
    private Button example2;
    private Button example3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        example1 = (Button) findViewById(R.id.example1);

        example2 = (Button) findViewById(R.id.example2);
        example2.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                Intent intent = new Intent().setClass(getApplicationContext(), UmbrellaActivity.class);
                startActivity(intent);
            }
        });

        example3 = (Button) findViewById(R.id.example3);
    }
}

File: app/src/main/java/mx/yellowme/app/MainActivity.kt
package mx.yellowme.app

import android.content.Intent
import kotlinx.android.synthetic.main.activity_main.*

import mx.yellowme.app.simple.UmbrellaActivity
import mx.yellowme.waterproof.mvp.activity.WpActivity

class MainActivity :
        WpActivity<MainPresenter>(),
        MainContract.View {

    override val layout: Int
        get() = R.layout.activity_main

    override fun setupActivity() {

        example2?.setOnClickListener {
            val intent = Intent().setClass(applicationContext, UmbrellaActivity::class.java)
            startActivity(intent)
        }
        presenter?.getDummyMessage()
    }

    override fun displayDummyMessage(message: String?) {
        display(message ?: "Mensaje nullo")
    }
}
